src/v/storage/tests/BUILD

```diff
@@ -0,0 +1,39 @@
+load("//bazel:test.bzl", "redpanda_cc_btest", "redpanda_cc_gtest")
+
+redpanda_cc_gtest(
+    name = "segment_offset_tracker_test",
+    timeout = "short",
```

Comment:
Is this pretty much the set of tests that has all of its dependencies bazelified?

Reply:
no, just enough to catch any linking issues. i think quite a few of the storage tests can be added now.